config/base.go

```diff
@@ -0,0 +1,81 @@
+// Copyright © 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+// BaseConfig is a generic interface for TOML configs in mixer
+type BaseConfig interface {
```

Comment:
In the effort of writing more usable code, I'd change `BaseConfig` to just `Config`. I am not a fan of **Base** `Base Classes` :) i.e,
```
class Dog {}
class Labrador : public Dog {} 
```

makes a lot more sense than 
```
class BaseDog{} 
class Labrador : public BaseDog{}
```

What is a BaseDog? :smiley:

Reply:
I agree, but the reason why I don't wanna call it "Config" is to avoid naming conflict. It happened when we had a global variable named "config" in `cmd/` and I also regret naming the file that handles builder conf `config/config.go`. on top of that we also have `swupd/config.go` that declares a structure named "Config".
So yeah, I don't wanna add yet another thing to the salad.

Reply:
`Config` and `config` are not naming conflicts in Go. If it helps to rename the files please feel free to, they don't all have to be named config.go, however since they're namespaced by package it won't cause any conflicts since they will be prefaced by `swupd.<etc> `and `config.<etc>` whenver they're used elsewhere.

Reply:
And that is the reason why the config inside `swupd` hasn't moved into the `config` package, which would make much more sense. Also the reason why `swapd` package doesn't use `config` package, since if conflicts with variables named `config`.

But ok, lets rename stuff:
- base.go -> config.go
- config.go -> mixconfig.go
- state.go -> mixstate.go
- BaseConfig -> Config

Sounds good?

Reply:
Sounds great :smile: